File: RobWork/example/python/ex-quaternion.py
from sdurw import *
import math

if __name__ == '__main__':
    quat = Quaternion(math.sqrt(2)/2,math.sqrt(2)/2,0,0)
    print("Quaternion: " + str(quat))
    rotationFromQuat = quat.toRotation3D()
    print("Rotation from Quaternion: " + str(rotationFromQuat));

    rot = Rotation3Dd(-1,0,0,0,0,1,0,1,0);
    print("Rotation: " + str(rot));
    quatFromRotation = Quaternion(rot);
    print("Quaternion from Rotation: " + str(quatFromRotation));

File: RobWork/example/lua/ex-quaternion.lua
require("sdurw_core")
require("sdurw")
using("sdurw_core")
require("sdurw_math")
using("sdurw_math")

if Quaternion == nil then 
  Quaternion = Quaternion_d
end

quat = Quaternion(math.sqrt(2)/2,math.sqrt(2)/2,0,0);
print("Quaternion: " .. tostring(quat))
rotationFromQuat = quat:toRotation3D();
print("Rotation from Quaternion: " .. tostring(rotationFromQuat));

rot = Rotation3Dd(-1,0,0,0,0,1,0,1,0);
print("Rotation: " .. tostring(rot));
quatFromRotation = Quaternion(rot);
print("Quaternion from Rotation: " .. tostring(quatFromRotation))
